### PR TITLE
Add the ability to set negotation for an interface

### PR DESCRIPTION
**Pull request by sc68cal**
##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: roles/os10_interface/templates/os10_interface.j2 [hunk 1]
```diff
@@ -123,7 +123,11 @@ interface {{ interface_key }}
  no speed
     {% endif %}
   {% endif %}
-
+  {% if intf_vars.auto_neg is defined %}
+    {% if intf_vars.auto_neg %}
+negotiation {{ intf_vars.auto_neg }}
```
Comment (gunasekaran-t) [inline roles/os10_interface/templates/os10_interface.j2:128]: Please add 'no' form of the CLI as well.

**Comment by gunasekaran-t** [comment]
Please add a changelog fragment file

**Comment by gunasekaran-t** [comment]
Please update document file as well.

**Comment by Andersson007** [comment]
@sc68cal hello, how about adding the fragment (see https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and updating the documentation?

**Comment by Andersson007** [comment]
cc @sc68cal ^

**Comment by sc68cal** [comment]
won't be able to get around to this for a bit

**Comment by Andersson007** [comment]
@sc68cal ok, no problem! Wanted to be sure that you saw this

**Comment by sc68cal** [comment]
@Andersson007 updated

**Comment by Andersson007** [comment]
@sc68cal cool, thanks, please unmark it as draft

**Comment by Andersson007** [comment]
cc @gunasekaran-t @dharmaraj-gurusamy if you're happy with this, please approve (and merge or I can merge)

**Comment by Andersson007** [comment]
@sc68cal thanks for the contribution!
@dharmaraj-gurusamy thanks for reviewing and merging!

**Comment by irgama** [comment]
This change breaks the interface role for any interface that does not have "negotiation" as an option. I.E. If you try to configure VLANs, then the role will fail at the VLAN. This is only now becoming relevant with this change present when you pull the collection from ansible galaxy.

**Pull request merged**